### PR TITLE
st1 [2131][IMP] stock_picking_comment_template_ext

### PR DESCRIPTION
**Pull request by tack-yabeQuartile**
[2131](https://www.quartile.co/web#id=2131&action=771&model=project.task&view_type=form&menu_id=505)

### CODE REVIEW

**Comment by tack-yabeQuartile** [comment]
As a function check (for myself) has done like below.

1:
* Copy a sale_order as quotation.
* Write something in Staff Memo for SO
* Store the quotation
* Push the "CONFIRM" button (to run action_confirm method)
* Open picking (transfer) form.
* Ensure that the Staff Memo for picking have the same content to the Staff Memo for SO
  → **staff memo for picking inherits the content for SO when picking(transfer) was created.**

2:
* Edit the content of Staff Memo for picking.
* Ensure that the Staff Memo for So is NOT edited.
* Edit the content of Staff Memo for SO (not the same to the content for picking)
* Ensure that the Staff Memo for picking is NOT changed.
  →  **staff memos are independent of each other.**

File: stock_picking_comment_template_ext/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.multi
+    def _prepare_procurement_values(self, group_id=False):
+        values = super(SaleOrderLine, self)._prepare_procurement_values(group_id)
+        self.ensure_one()
+        values.update({"note2": group_id.sale_id.note2})
+        return values
```
Comment (yostashiro) [inline stock_picking_comment_template_ext/models/sale_order.py]: I guess we don't need this any more.

File: stock_picking_comment_template_ext/models/stock_picking.py [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    note2 = fields.Html(compute="_compute_note2",)
-
-    @api.multi
-    def _compute_note2(self):
-        for picking in self:
-            picking.note2 = picking.group_id.sale_id.note2
+    note2 = fields.Html(default=lambda self: self.group_id.sale_id.note2)
```
Comment (yostashiro) [inline stock_picking_comment_template_ext/models/stock_picking.py]: Does this lambda actually work?  If it does, then no need to extend _get_new_picking_values() in stock.move.

File: stock_picking_comment_template_ext/README.rst [hunk 3]
```diff
@@ -14,16 +14,16 @@ Picking Comments Ext
     :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-qrtl%2Frpl--custom-lightgray.png?logo=github
-    :target: https://github.com/qrtl/rpl-custom/tree/12.0/stock_picking_comment_template_ext
+    :target: https://github.com/qrtl/rpl-custom/tree/13.0/stock_picking_comment_template_ext
```
Comment (yostashiro) [inline stock_picking_comment_template_ext/README.rst]: Revert the version to 12.0.

File: stock_picking_comment_template_ext/README.rst [hunk 4]
```diff
@@ -14,16 +14,16 @@ Picking Comments Ext
...
 
 |badge1| |badge2| |badge3| 
 
 This module does the following:
 
 * Changes the string "Bottom Comments" to "Staff Memo".
-* Makes Staff Memo (note2) of picking a computed field, always mirroring the
-  content from the sales order.
+* Only when picking is created, the memo for sale_order is copied to one for picking as the initial value.
+* These contents are independent of each other.
```
Comment (yostashiro) [inline stock_picking_comment_template_ext/README.rst]:
```suggestion
* When picking is created from sales order, the value of memo (note2) field of the sales order is copied to the corresponding field of the picking.
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: stock_picking_comment_template_ext/README.rst (hunk 3 above)
Comment (tack-yabeQuartile) [reply stock_picking_comment_template_ext/README.rst]: OK, I do. (The other repo's settings were left unchanged.)

File: stock_picking_comment_template_ext/models/stock_picking.py (hunk 2 above)
Comment (tack-yabeQuartile) [reply stock_picking_comment_template_ext/models/stock_picking.py]: I will check again. 
Without both _prepare_procurement_values and _get_new_picking_values(), this did not work in this morning.

File: stock_picking_comment_template_ext/models/sale_order.py (hunk 1 above)
Comment (tack-yabeQuartile) [reply stock_picking_comment_template_ext/models/sale_order.py]: OK I see. I will remove after check the behavior.

**Comment by tack-yabeQuartile** [comment]
now, I'm checking the behavior of lambda function.
2 patterns were verified.

First, comment out `from . import stock_move` in models/__ini__.py
Result：Contents of memo was not inherited.

Second, remove `default` option from `note2 = fields.Html()`
Result：Contents of memo was inherited successhully.

→default option does not work and no need to write.

File: stock_picking_comment_template_ext/models/stock_move.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _get_new_picking_values(self):
+        vals = super(StockMove, self)._get_new_picking_values()
+        vals["note2"] = self.sale_line_id.order_id.note2
```
Comment (yostashiro) [inline stock_picking_comment_template_ext/models/stock_move.py]: @tack-yabeQuartile Can you check if this doesn't hinder the purchase order confirmation where sale_line_id of the move is not updated.

File: stock_picking_comment_template_ext/models/stock_picking.py (hunk 2 above)
Comment (yostashiro) [reply stock_picking_comment_template_ext/models/stock_picking.py]: > Without both _prepare_procurement_values and _get_new_picking_values(), this did not work in this morning.

That doesn't make sense to me. You should only need either.

Comment (tack-yabeQuartile) [reply stock_picking_comment_template_ext/models/stock_picking.py]: Oops, I meant "Without neither the two methods,  default=lambda did not work."

Comment (tack-yabeQuartile) [reply stock_picking_comment_template_ext/models/stock_picking.py]: now, We know only _get_new_picking_values() is needed and `default=`  is not required.

**Review by yostashiro — APPROVED** [review]
Code review. LGTM 👍

**Comment by tack-yabeQuartile** [comment]
tests/test_stock_picking_comment.py has been removed since this test does not match the customer's requirements.
After remove this test file, unit test was passed in my local env.

**Comment by tack-yabeQuartile** [comment]
More careful check is needed to adjust test files. I will check the unit test again and build another PR.

**Pull request merged**